### PR TITLE
Adds update function for pending totals

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/2890

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
- enable rewards
- in current window open panel
- open new window with rewards settings page
- open new window with unverified site
- tip that site
- make sure that panel and settings page display the amount

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -2200,8 +2200,19 @@ bool SavePendingContributionOnFileTaskRunner(PublisherInfoDatabase* backend,
...
 
+void RewardsServiceImpl::OnCallBackPendingContribution(double amount) {
+  for (auto& observer : observers_)
+    observer.OnPendingContributionsTotal(this, amount);
+}
+
 void RewardsServiceImpl::OnSavePendingContribution(bool result) {
-  // TODO(nejczdovc) add callback with db result
+  if (result) {
+    GetPendingContributionsTotal(base::Bind(
```
Comment (bridiver) [inline components/brave_rewards/browser/rewards_service_impl.cc]: we shouldn't be making queries to update the UI here. This should trigger a PendingContributionSaved method on the observer and the UI should call GetPendingContributionsTotal in response to that

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 2]
```diff
@@ -2200,8 +2200,19 @@ bool SavePendingContributionOnFileTaskRunner(PublisherInfoDatabase* backend,
   return backend->InsertPendingContribution(list);
 }
 
+void RewardsServiceImpl::OnCallBackPendingContribution(double amount) {
+  for (auto& observer : observers_)
+    observer.OnPendingContributionsTotal(this, amount);
+}
+
 void RewardsServiceImpl::OnSavePendingContribution(bool result) {
```
Comment (bridiver) [inline components/brave_rewards/browser/rewards_service_impl.cc]: as a side note results shouldn't be `bool` they should use result codes so we can provide additional info

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc]: database return just true or false, so not sure what additional info we can get

Comment (bridiver) [reply components/brave_rewards/browser/rewards_service_impl.cc]: I think it's still better for consistency to just use Result everywhere in rewards service

Comment (bridiver) [reply components/brave_rewards/browser/rewards_service_impl.cc]: and something in the backend could change to potentially respond with more detailed errors

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc]: done

**Pull request merged**